### PR TITLE
[#720][FOLLOW-UP] Correct the shuffle server id

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

Assign the correct shuffle server id

### Why are the changes needed?
#720 follow-up pr

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing UTs

### CODE REVIEW

**Comment by xumanbu** [comment]
I have an idea for `ShuffleServerId`, and `new ShuffleFlushManager with Future<String> shuffleServerId`
```
public class ShuffleServerIdCreator {

  private String ip;
  private Integer grpcPortConf;
  private Integer nettyPortConf;

  private final ExecutorService executor = Executors.newSingleThreadExecutor();
  private final SynchronousQueue<Integer> nettyPortFuture = new SynchronousQueue();

  public ShuffleServerIdCreator(String ip, Integer grpcPortConf, Integer nettyPortConf) {
    this.ip = ip;
    this.grpcPortConf = grpcPortConf;
    this.nettyPortConf = nettyPortConf;
  }

  public Future<String> create() {
    return executor.submit(() -> {
      if (nettyPortConf > 0) {
        return ip + "-" + grpcPortConf + "-" + nettyPortConf;
      } else if (nettyPortConf == 0) {
        return ip + "-" + grpcPortConf + "-" + nettyPortFuture.take();
      } else {
        return ip + "-" + grpcPortConf;
      }
    });
  }

  public void setNettyActualPort(int actualNettyPort) {
    nettyPortFuture.offer(actualNettyPort);
  }
}
```

**Comment by jerqi** [comment]
> I have an idea for `ShuffleServerId`, and `new ShuffleFlushManager with Future<String> shuffleServerId`
> 
> ```
> public class ShuffleServerIdCreator {
> 
>   private String ip;
>   private Integer grpcPortConf;
>   private Integer nettyPortConf;
> 
>   private final ExecutorService executor = Executors.newSingleThreadExecutor();
>   private final SynchronousQueue<Integer> nettyPortFuture = new SynchronousQueue();
> 
>   public ShuffleServerIdCreator(String ip, Integer grpcPortConf, Integer nettyPortConf) {
>     this.ip = ip;
>     this.grpcPortConf = grpcPortConf;
>     this.nettyPortConf = nettyPortConf;
>   }
> 
>   public Future<String> create() {
>     return executor.submit(() -> {
>       if (nettyPortConf > 0) {
>         return ip + "-" + grpcPortConf + "-" + nettyPortConf;
>       } else if (nettyPortConf == 0) {
>         return ip + "-" + grpcPortConf + "-" + nettyPortFuture.take();
>       } else {
>         return ip + "-" + grpcPortConf;
>       }
>     });
>   }
> 
>   public void setNettyActualPort(int actualNettyPort) {
>     nettyPortFuture.offer(actualNettyPort);
>   }
> }
> ```

A little complex. I fix this issue in another way. You can take a look.

File: server/src/main/java/org/apache/uniffle/server/ShuffleServer.java [hunk 1]
```diff
@@ -125,13 +125,21 @@ public static void main(String[] args) throws Exception {
...
 
+    if (nettyServerEnabled) {
+      id = ip + "-" + grpcPort + "-" + nettyPort;
+    } else {
+      id = ip + "-" + grpcPort;
+    }
+    shuffleServerConf.setString(ShuffleServerConf.SHUFFLE_SERVER_ID, id);
+    LOG.info("Start to initialize server {}", id);
+
+    registerHeartBeat.startHeartBeat();
```
[the remark this answers is not in the document]
Comment (jerqi) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServer.java:142]: We call `registerMetrics` in the method `initialization`.

[the remark this answers is not in the document]
Comment (jerqi) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServer.java:143]: fixed.

**Review by xianjingfeng — APPROVED** [review]

**Pull request merged**